docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+      name: user.name,
+    },
+    loggedInAt: new Date(),
+  });
+});
+
+
+```
+
+The user should now be logged in! With the session set, we can get the current user session in any API route or page by calling `getUserSession(event)` which is auto-imported as a util function from the `nuxt-auth-utils` package.
+
+### 6. Create a logout API route
+
+Users need to be able to log out, so we should create an API route to allow them to do this. This should require post request as well, just like login. We'll clear the session when this endpoint is called.
+
+We'll use the `clearUserSession` from the `auth-utils` module to log the user out and clear the session data. This function is automatically imported from the module by Nuxt, so we don't need to manually import it.
+
+```typescript [server/api/auth/logout.post.ts]
+export default defineEventHandler(async (event) => {
+  // Clear the current user session
+  await clearUserSession(event);
+});
+```
+
+### 7. Protect your server route
+
+Protecting server routes is key to making sure your data are safe. Front-end middleware is helpful for the user, but without back-end protection your data can still be accessed. Because of this, it is critical that we protect any API routes with sensitive data. For these sensitive routes, we should return a 401 error if the user is not logged in.
+
+The `auth-utils` module provides the `requireUserSession` utility function to help make sure that users are logged in and have an active session. We can use this to protect our different endpoints. Like many of the other utilities from the auth module, it is automatically imported in our server endpoints.
+
+In the example below, we use the `requireUserSession` utility function to protect the `/server/api/users.get.ts` server route. This route will only be accessible to logged-in users.
+
+```typescript [/server/api/users.get.ts]
+import getDatabase from "~/database/database";
+import users from "~/database/schema/users";
+import requireUserLoggedIn from "~/server/utils/requireUserLoggedIn";
+
+export default defineEventHandler(async (event) => {
+  // make sure the user is logged in
+  // This will throw a 401 error if the request doesn't come from a valid user session
+  await requireUserSession(event);
+
+  // If we make it hear, the user is authenticated. It's safe to fetch and return data
+  const db = await getDatabase();
+  
+  // Send back the list of users
+  const userList = await db.select({ name: users.name, id: users.id }).from(users).limit(10);
+
+  return userList;
+});
+```
+
+### 8. Create a front-end middleware to protect routes
+
+Our data are safe with the back-end route in place, but without doing anything else, unauthenticated users would probably get some odd data when trying to access the `/users` page. We should create a [front-end middleware](https://nuxt.com/docs/guide/directory-structure/middleware) to protect the route on the client side and redirect users to a login page.
```

Comment:
Data is safe here as well, and the same suggestion about back-end and front-end

"redirect users to a login page." should use `the` in stead of `a`, as there is only one specific login page:
"redirect users to the login page."

Reply:
I agree that grammatically it is correct, but the nuance is about finding the right amount of compromise between grammar and beginner-friendliness, similar to front-end and back-env vs client-side and server-side. Though all in all, I believe this could stay as is as it doesn't change much

Reply:
Alright. I wanted to make sure this was taken into consideration. I've pushed up the language change since you think it'll be easier for readers.

Reply:
Sounds good to me! Even if you do choose to revert it in the end it would be totally fine, as this has very little impact on the recipe after all.